docs/source/user_manual/tutorial_van_der_waal.rst

```diff
@@ -167,26 +174,48 @@ Next, there are two programing tasks to complete:
 The following is the code for these two needs::
 
     # Re-calculate when attraction, tot_volume, or temperature are changed.
-    @on_trait_change('attraction, tot_volume, temperature')
-    def calc(self):
+    @observe('attraction, tot_volume, temperature')
+    def calc(self, event=None):
         """ Update the data based on the numbers specified by the user. """
         self.volume = arange(.1, 100)
         self.pressure = (
             (self.r_constant*self.temperature)/(self.volume - self.tot_volume)
             - self.attraction/(self.volume*self.volume)
         )
 
+        self.plot.data.set_data("x", self.volume)
+        self.plot.data.set_data("y", self.pressure)
+
 The :func:`calc` function computes the :attr:`pressure` array using the current
-values of the independent variables.  Meanwhile, the
-:func:`@on_trait_change` decorator (provided by Traits) tells Python to call
+values of the independent variables. It then updates the :attr:`data` of our
+:attr:`plot` to use the newly computed values.  Meanwhile, the
+:func:`@observe` decorator (provided by Traits) tells Python to call
 :func:`calc` whenever any of the attributes :attr:`attraction`,
 :attr:`tot_volume`, or :attr:`temperature` changes.
 
+In addition to reacting to changes in the input parameters, we also want our
+plot to change based on the user selected :attr:`plot_type`.  To do this, we can
+define a separate listener as follows::
+
+    @observe("plot_type")
+    def _update_plot_type(self, event):
+        old_plot_type, new_plot_type = event.old, event.new
+
+        self.plot.delplot(old_plot_type)
+        self.plot.plot(
+            ("x", "y"), type=new_plot_type, name=new_plot_type, color="blue"
+        )
```

Comment:
so when i change `plot_type`, i need to change some other variable before the plot is updated i.e. if i change from line to scatter, i need to change temperature before the plot changes from line plot to scatter plot.

Reply:
Ah I wasn't observing this locally.  Thanks for pointing this out. I tried adding `self.plot.invalidate_and_redaw()`. Lmk if this fixes it for you

Reply:
FWIW I was seeing something similar to what you describe while working on the hyetograph example and `invalidate_and_redraw` fixed it